zerver/tests/test_signup.py

```diff
@@ -935,7 +935,7 @@ def test_register(self) -> None:
         # seem to be any O(N) behavior.  Some of the cache hits are related
         # to sending messages, such as getting the welcome bot, looking up
         # the alert words for a realm, etc.
-        with self.assert_database_query_count(103), self.assert_memcached_count(18):
+        with self.assert_database_query_count(104), self.assert_memcached_count(18):
```

Comment:
This addition in query is for accessing `realm.can_access_all_users_group.name`. Since the realm object is obtained from `PreregistrationUser` obejct, there is no simple way of using `select_related` currently and may require some more refactoring.

Reply:
It's probably fine to just have this one extra query for user creation. But please add a `TODO` comment about this detail just before the `user_access_restricted_in_realm` call in this commit noting that it ends up doing a database query because the queries to fetch `PreregistrationUser` do not prefetch it.